manager/container_test.go

```diff
@@ -319,3 +321,132 @@ func TestOnDemandHousekeepingReturnsAfterStopped(t *testing.T) {
+	[]byte("root         104       2 21:34  0.0  0.0     0      0 I<   00:00:00 kthrotld          3 -\nroot         105       2 21:34  0.0  0.0     0      0 S    00:00:00 irq/41-aerdrv     0 -\nroot         107       2 21:34  0.0  0.0     0      0 I<   00:00:00 DWC Notificatio   3 -\nroot         109       2 21:34  0.0  0.0     0      0 S<   00:00:00 vchiq-slot/0      1 -\nroot         110       2 21:34  0.0  0.0     0      0 S<   00:00:00 vchiq-recy/0      3 -"),
+}
+
+func TestParseProcessList(t *testing.T) {
+	for i, ps := range psOutput {
+		t.Run(fmt.Sprintf("iteration %d", i), func(tt *testing.T) {
+			cd := &containerData{}
+			_, err := cd.parseProcessList("/", true, ps)
+			// checking *only* parsing errors - otherwise /proc would have to be emulated.
+			assert.NoError(tt, err)
+		})
+	}
+}
+
+var psLine = []struct {
+	name              string
+	line              string
+	cadvisorContainer string
+	isHostNamespace   bool
+	process           *v2.ProcessInfo
+	err               error
+	cd                *containerData
+}{
+	{
+		"plain process with cgroup",
+		"ubuntu     15888    1804 23:51  0.1  0.0  2832  10176 R+   00:10:00 cadvisor            1 10:cpuset:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,9:devices:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,8:pids:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,7:memory:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,6:freezer:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,5:perf_event:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,4:blkio:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,3:cpu,cpuacct:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,2:net_cls,net_prio:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831,1:name=systemd:/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831",
+		"/docker/cadvisor",
+		true,
+		&v2.ProcessInfo{
+			User:          "ubuntu",
+			Pid:           15888,
+			Ppid:          1804,
+			StartTime:     "23:51",
+			PercentCpu:    0.1,
+			PercentMemory: 0.0,
+			RSS:           2899968,
+			VirtualSize:   10420224,
+			Status:        "R+",
+			RunningTime:   "00:10:00",
+			CgroupPath:    "/docker/dd479c33249f6c3f0f1189aa88f07dad3eeb3e6fedfc71385c27ddd699994831",
+			Cmd:           "cadvisor",
+			Psr:           1,
+		},
+		nil,
+		&containerData{
+			info: containerInfo{ContainerReference: info.ContainerReference{Name: "/"}},
+		},
+	},
+	{
+		"process with space in name and no cgroup",
+		"root         107       2 21:34  0.0  0.1     3      4 I<   00:20:00 DWC Notificatio   3 -",
+		"/docker/cadvisor",
+		false,
```

Comment:
nit: i think it would be a bit more readable if you included the struct names for each field inline here, since it can be a bit confusing what the struct fields map too.